back/engines/commercial/verification/app/services/verification/patches/permissions_service.rb

```diff
@@ -12,6 +12,23 @@ def denied_when_permitted_by_groups?(permission, user)
 
         denied_reason
       end
+
+      private
+
+      def base_requirements(permission)
+        requirements = super
+        if VerificationService.new.find_verification_group(permission.groups)
+          requirements['groups'][:special][:verification] = 'require'
+        end
+        requirements
+      end
+
+      def mark_satisfied_requirements!(requirements, permission, user)
+        super
+        return unless user.verified? && VerificationService.new.find_verification_group(permission.groups)
```

Comment:
I'm not sure this is correct. Normally the groups are ignored, unless permitted_by is set to "groups".

Reply:
I think maybe I'll need to check if permitted_by is set to groups before doing this